share/pvss/pvss.go

```diff
@@ -159,13 +172,36 @@ func VerifyDecShare(suite Suite, G kyber.Point, X kyber.Point, encShare *PubVerS
 
 // VerifyDecShareBatch provides the same functionality as VerifyDecShare but for
 // slices of decrypted shares. The function returns the the valid decrypted shares.
-func VerifyDecShareBatch(suite Suite, G kyber.Point, X []kyber.Point, encShares []*PubVerShare, decShares []*PubVerShare) ([]*PubVerShare, error) {
-	if len(X) != len(encShares) || len(encShares) != len(decShares) {
+func VerifyDecShareBatch(suite Suite, G kyber.Point, pubRun *PubRun, decShares []*PubVerShare, cExp kyber.Scalar) ([]*PubVerShare, error) {
+	// match index to encShare and decShare
+	indToShare := make(map[int]*PubVerShare, pubRun.N)
+	indToPk := make(map[int]kyber.Point, pubRun.N)
+	for i, el := range pubRun.Shares {
+		indToShare[el.S.I] = el
+		indToPk[el.S.I] = pubRun.X[i]
+	}
+
+	filteredDecShares := make([]*PubVerShare, 0)
+	filteredEncShares := make([]*PubVerShare, 0)
+	filteredPks := make([]kyber.Point, 0)
+	for _, decShare := range decShares {
+		if encShare, ok := indToShare[decShare.S.I]; ok {
```

Comment:
is there a use case for returning valid decrypted shares but silently ignoring that some are invalid?

Reply:
If enough are valid we can still retrieve the secret, otherwise the user would need to do the filtering themselves, we also don't have a way to log the event